File: sentimental-readability/readability.py
def main():
    text = input("Text: ")
    X = index(text)
    if X<1:
        print("Before Grade 1")
    elif X<16:
        print("Grade",X)
    else:
        print("Grade 16+")

def index(text):
    L=0
    S=0
    W=1

    for c in text:
        if c.isalpha():
            L+=1
        elif c.isspace():
            W+=1
        elif c in ['.','!','?']:
            S+=1

    L = (L/W)*100
    S = (S/W)*100

    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)

if __name__=="__main__":
    main()

File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int letterCount(string text, int length);
int wordCount(string text, int length);
int sentenceCount(string text, int length);

int main(){
    string text = get_string("Text: ");
    int length = strlen(text);

    int letters = letterCount(text,length);
    int words = wordCount(text,length);
    int sentences = sentenceCount(text,length);

    double L = (letters/(double)words)*100;
    double S = (sentences/(double)words)*100;
    double index = 0.0588 * L - 0.296 * S - 15.8;

    int X = round(index);

    if(X>16){
        printf("Grade 16+\n");
    }
    else if(X<1){
        printf("Before Grade 1\n");
    }
    else{
        printf("Grade %i\n",X);
    }
    return 0;
}

int letterCount(string text, int length){
    int count = 0;
    for(int i=0; i<length; i++){
        if(isalpha(text[i])){
            count++;
        }
    }
    return count;
}
int wordCount(string text, int length){
    int count=1;
    for(int i=0; i<length; i++){
        if(text[i]==' '){
            count++;
        }
    }
    return count;
}
int sentenceCount(string text, int length){
    int count=0;
    for(int i=0; i<length; i++){
        if(text[i]=='.'|| text[i]=='!' || text[i]=='?'){
            count++;
        }
    }
    return count;
}
